stl/inc/chrono

```diff
@@ -2209,6 +2221,2949 @@ namespace chrono {
+            auto _Optional           = _F_wkday | _F_doy;
+            if (_For_time_point) {
+                // These fields aren't used here, but they might be used later if converting to a time_point.
+                _Optional |= _F_sec | _F_min | _F_hr;
+            }
+
+            if (_Consistent && _Test_bits(_Used_fields(), _Required, _Optional)) {
+                _Year_month_day_result = year_month_day{year{*_Century + *_Two_dig_year},
+                    month{static_cast<unsigned int>(*_Month)}, day{static_cast<unsigned int>(*_Day)}};
+                return true;
+            } else {
+                return false;
+            }
+        }
+
+        template <class _InIt>
+        _NODISCARD _InIt _Parse_time_field(_InIt _First, ios_base& _Iosbase, ios_base::iostate& _State,
+            const char _Flag, const char _Modifier, const unsigned int _Width,
+            const unsigned int _Subsecond_precision) {
+            using _CharT = typename _InIt::value_type;
+
+            const auto& _Ctype_fac = _STD use_facet<ctype<_CharT>>(_Iosbase.getloc());
+            const auto& _Time_fac  = _STD use_facet<time_get<_CharT>>(_Iosbase.getloc());
+            constexpr _InIt _Last{};
+
+            int _Val{0};
+            switch (_Flag) {
+            case 'a':
+            case 'A':
+                {
+                    tm _Tp;
+                    _Tp.tm_wday = _Invalid_time_field<int>;
+                    _First      = _Time_fac.get(_First, _Last, _Iosbase, _State, &_Tp, 'a');
+                    if (!_Update(_Weekday, _Tp.tm_wday)) {
+                        _State |= ios_base::failbit;
+                    }
+                    break;
+                }
+
+            case 'b':
+            case 'B':
+            case 'h':
+                {
+                    tm _Tp;
+                    _Tp.tm_mon = _Invalid_time_field<int>;
+                    _First     = _Time_fac.get(_First, _Last, _Iosbase, _State, &_Tp, 'b');
+                    if (_Tp.tm_mon == _Invalid_time_field<int> || !_Update(_Month, ++_Tp.tm_mon)) {
+                        _State |= ios_base::failbit;
+                    }
+                    break;
+                }
+
+            case 'C':
+                if (_Modifier != 'E') {
+                    _State |= _Get_int(_First, _Width == 0 ? 2u : _Width, _Val, _Ctype_fac);
+                    _Val *= 100;
+                } else {
+                    tm _Tp;
+                    _Tp.tm_year = _Invalid_time_field<int>;
+                    _First      = _Time_fac.get(_First, _Last, _Iosbase, _State, &_Tp, 'C', 'E');
+                    _Val        = _Tp.tm_year;
+                    if (_Tp.tm_year != _Invalid_time_field<int>) {
+                        _Val += 1900;
+                    }
+                }
+
+                if (!_Update(_Century, _Val)) {
+                    _State |= ios_base::failbit;
+                }
+                break;
+
+            case 'd':

```

Comment:
This is C++20 so should we add `[[fallthrough]]`?

I am not 100% sure whether a compiler would warn here, given that the cases are identical

Reply:
While `case 'd'` is *technically* falling through to `case 'e'`, I am confident that no compilers (going back to C) have ever warned about cases being grouped together in this manner, when they are totally empty. Fallthrough is only dangerous (and in need of the `[[fallthrough]]` attribute) when a case performs some work and then falls through to the next.

I would be strongly opposed to adding the attribute here - not only would it add noise, it would reduce the usefulness of `[[fallthrough]]` as a rare warning sign that something unusual is happening.

Reply:
Hm i should get myself a *tongue-in-cheek* emoji for not so serious review comments